Use new sql interpolator in more DAOs, clean up

fm3 opened the pull request:
- propagates the use of the new sql interpolator with q"myQuery" that was introduced in #6718 to more DAOs
- also adapts the types of the access queries (meaning the DAOs that are NOT yet adapted must re-stringify the access queries using `.debugInfo`
- splits classes of the file SQLHelpers into individual files
- adds support for enums, boundingboxes, vec3ints, arrays in sql interpolator

### URL of deployed dev instance (used for testing):
- https://sqlrefactoring.webknossos.xyz

### Issues
- contributes to #6721

### TODO
 - [x] split SQLHelpers file, one file per class
 - [x] make access queries sqlTokens
 - [x] solve setting custom enumerations
 - [x] fix setting varchar[] (e.g. annotation tags)
 - [x] fix setting nulled jsonb
 - [x] fix setting Vec3Double
 - [x] check that error logging does not leak user data

### Follow-up
 - adapt all remaining queries
 - remove then-unused escaping/unescaping code
 - remove then-unused SqlTypeImplicits trait
 - check that no usages of #$ remain
 - check that no usages of debugInfo remain
 - check that no usages of sql"" and sqlu"" remain, remove their import
 - replace the Any/match approach to an implicit conversion approach to re-enable type checks at compile time

### Steps to test:
- Test creating/importing/updating/deleting datasets, annotations, private links, folders
- Test with multi-user setup that access checks still work (e.g. dataset list, with folder permissions)
- CI helps as well
- do some monkey-testing of things that I am not thinking of :sweat_smile: 

------
- [x] Ready for review

fm3 commented:
@normanrz Could you have a look at the changes in the SqlInterpolation file, and possibly check a few usages to see if I use this the way you intended it? @jstriebel agreed to look at the other changes :) It would also be appreciated if you could do some random testing. The fact that all SQL errors are logged and reported to slack makes me optimistic that we should find errors quickly

normanrz commented on app/utils/sql/SQLDAO.scala:
```diff
@@ -0,0 +1,99 @@
+
+  protected def updateStringCol(id: ObjectId, column: X => Rep[String], newValue: String)(
+      implicit ctx: DBAccessContext): Fox[Unit] = {
+    val q = for { row <- collection if notdel(row) && idColumn(row) === id.id } yield column(row)
```
`val q` shadows the `q`-query builder

fm3 replied:
Good catch, I renamed these to query. I am thinking of removing those entirely in the near future, though, to have only one unified way of creating sql queries.

fm3 commented on app/models/binary/DataSet.scala:
```diff
@@ -116,23 +113,25 @@ class DataSetDAO @Inject()(sqlClient: SQLClient,
-            WHERE apl.accessToken = '$t'
-          )""").getOrElse("")
+            WHERE apl.accessToken = $t
+          )""").getOrElse(q"false")
```
values seems better, I’ve adapted the queries now :)

fm3 commented:
Thanks for your feedback!
 - I changed the boolean constants to pass values
 - I added tests
 - I hope to adapt the remaining DAOs in the coming days, but I think a separate PR for that would make sense. There are still more than 100 queries to be adapted.
 - I created follow-up issue #6730  for the helper functions, as it is not immediately obvious to me how to design them

normanrz commented on app/models/annotation/Annotation.scala:
```diff
@@ -366,51 +365,48 @@ class AnnotationDAO @Inject()(sqlClient: SQLClient, annotationLayerDAO: Annotati
-         ${a._user}, ${a.description}, '#${a.visibility}', ${a.name},
-         #${optionLiteral(viewConfigurationStr.map(sanitize))},
-         '#${a.state}', '#${sanitize(a.statistics.toString)}',
-         '#${writeArrayTuple(a.tags.toList)}', ${a.tracingTime}, '#${a.typ}',
+         ${a._user}, ${a.description}, ${a.visibility}, ${a.name},
+         ${a.viewConfiguration},
+         ${a.state}, ${a.statistics},
+         ${a.tags}, ${a.tracingTime}, ${a.typ},
```
This would be neat:
```
SqlToken.insert("webknossos.annotations")(
  "_id" -> a._id,
  "_dataSet" -> a._dataSet,
  ...
).sql == "INSERT INTO webknossos.annotations(_id, _dataSet, ...) VALUES (?,?,...)"
```

jstriebel reviewed (approved):
LGTM, all things that looked weird are already part of the follow-up points :+1: :rocket: 
Props for adding tests! :heart:
